.github/workflows/linux.yml

```diff
@@ -129,9 +129,9 @@ jobs:
           # github.ref_name is 'ref/PR_#' in case of the PR, and 'branch_name' when executed on push
           save: ${{ github.ref_name == 'master' && 'true' || 'false'  }}
           verbose: 2
-          key: ${{ github.job }}-linux
+          key: linux-ubuntu
```

Comment:
just an option, we can use runner's info:
cache-${{ runner.os }}-${{ steps.system-info.outputs.platform}}

Reply:
All the GHA Linux pipelines are using the same runner group (`ubuntu-20.04-*`) but have different `cmake` commands and/or install different packages (building for different OSes/arches) so the `${{ runner.os }}` would be the same for all current/potential GHA pipelines.